File: src/main/java/org/shark/example/service/cassandra/SaveMessageService.java
package org.shark.example.service.cassandra;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.shark.example.datasource.cassandra.MessageRepository;
import org.shark.example.datasource.cassandra.pojo.MessageDo;
import org.shark.example.datasource.cassandra.pojo.MessageKeyDo;
import org.springframework.data.cassandra.core.CassandraTemplate;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;
import java.util.concurrent.CompletableFuture;

@Slf4j
@RequiredArgsConstructor
@Service
public class SaveMessageService {

    //    private final MessageRepository messageRepository;
    private final CassandraTemplate cassandraTemplate;

    public void saveMessage() {
        long userCount = 100;
        long messageCount = 10;

        List<MessageDo> messageList = new ArrayList<>();
        for (long i = 0; i < userCount; i++) {
            for (long j = 0; j < messageCount; j++) {
                MessageDo messageDo = new MessageDo();
                MessageKeyDo messageKeyDo = new MessageKeyDo();
                messageKeyDo.setUserId(i);
                messageKeyDo.setUserService("CUSTOMER");
                messageKeyDo.setItemId(j);
                messageDo.setMessageKey(messageKeyDo);
                messageDo.setTime(new Date());
                messageList.add(messageDo);
            }
        }
        long startTime = System.currentTimeMillis();


        cassandraTemplate.batchOps().insert(messageList).execute();

        long endTime = System.currentTimeMillis();
        log.info("total save time: {} ms", (endTime - startTime));
    }
}

File: src/main/kotlin/org/shark/example/service/cassandra/SaveMessageService.kt
package org.shark.example.service.cassandra

import org.shark.example.datasource.cassandra.pojo.MessageDo
import org.shark.example.datasource.cassandra.pojo.MessageKeyDo
import org.slf4j.LoggerFactory
import org.springframework.data.cassandra.core.CassandraTemplate
import org.springframework.stereotype.Service
import java.util.*

@Service
class SaveMessageService(
    private val cassandraTemplate: CassandraTemplate
) {
    private val log = LoggerFactory.getLogger(SaveMessageService::class.java)

    fun saveMessage() {
        val userCount: Long = 100
        val messageCount: Long = 10

        val messageList = ArrayList<MessageDo>()
        for (i in 0 until userCount) {
            for (j in 0 until messageCount) {
                val messageDo = MessageDo()
                val messageKeyDo = MessageKeyDo()
                messageKeyDo.userId = i
                messageKeyDo.userService = "CUSTOMER"
                messageKeyDo.itemId = j
                messageDo.messageKey = messageKeyDo
                messageDo.time = Date()
                messageList.add(messageDo)
            }
        }
        val startTime = System.currentTimeMillis()

        cassandraTemplate.batchOps().insert(messageList).execute()

        val endTime = System.currentTimeMillis()
        log.info("total save time: {} ms", (endTime - startTime))
    }
}
